staging/src/k8s.io/kubectl/pkg/cmd/apiresources/apiresources.go

```diff
@@ -208,44 +238,9 @@ func (o *APIResourceOptions) RunAPIResources() error {
 		}
 	}
 
-	if o.NoHeaders == false && o.Output != "name" {
-		if err = printContextHeaders(w, o.Output); err != nil {
-			return err
-		}
-	}
-
 	sort.Stable(sortableResource{resources, o.SortBy})
```

Comment:
Similar to my previous comment, I think this can be moved inside `APIResourcesPrinter` because it does not apply when you are printing JSON or YAML.

Reply:
I think we need to keep it here, since there's no straightforward way of accessing the `o.SortBy` within `APIResourcePrinter` as it is right now. Of course, we could modify `APIResourcePrinter` to also include the `sortBy` field and then this could go there. I'll do as you suggest.

Reply:
Yeah, I think sorting would be an implementation detail of the printer, so it makes sense to have a `sortBy` field on the printer.

Reply:
I think, there is no need sorting for json and yaml. We can add validation to not allow this and sorting can be moved under custom (default) printer.